### PR TITLE
Fix zappy compatibility for clip_array

### PR DESCRIPTION
**Pull request by flying-sheep**
- [ ] Closes #
- [ ] Tests included or not required because:

- [ ] Release notes not necessary because:

### CODE REVIEW

File: src/scanpy/preprocessing/_scale.py [hunk 1]
```diff
@@ -203,25 +217,13 @@ def scale_array(
...
         else:
-            if isinstance(X, DaskArray):
-                X = X.map_blocks(
-                    clip_array, max_value=max_value, zero_center=zero_center
-                )
-            elif issparse(X):
-                X.data = clip_array(X.data, max_value=max_value, zero_center=False)
-            else:
-                X = clip_array(X, max_value=max_value, zero_center=zero_center)
+            X = clip_array(X, max_value=max_value, zero_center=zero_center)
```
Comment (ilan-gold) [inline src/scanpy/preprocessing/_scale.py:226]: This works with zappy? is that the fix?

**Review by ilan-gold — APPROVED** [review]
Don't know much about zappy but it's certainly cleaner

File: src/scanpy/preprocessing/_scale.py (hunk 1 above)
Comment (flying-sheep) [reply src/scanpy/preprocessing/_scale.py:226]: Ah sorry; context.

No, the logic is untouched, only the `maybe_dask_process_context` is new.

This PR is part of me figuring out the whole “numba abort” situation, and this logic refactor is a side effect. I should probably retitle this PR.

Comment (ilan-gold) [reply src/scanpy/preprocessing/_scale.py:226]: "This" being this line that was the `else` with `clip_array` which appears to only be for `numpy`

Comment (flying-sheep) [reply src/scanpy/preprocessing/_scale.py:226]: I think that was another casualty of the flaky numba-on-M1 thing. Looking at it now, the logic seems identical.

**Pull request merged**